galsim/base.py

```diff
@@ -1462,10 +1462,12 @@ def drawKImage(self, re=None, im=None, nx=None, ny=None, bounds=None, scale=None
 
         return re,im
 
-    # Quick and dirty.  Just check reprs are equal.
-    def __eq__(self, other): return repr(self) == repr(other)
+    def __eq__(self, other):
```

Comment:
I guess it's a judgement call, but I think those two objects should still be considered equal.  Anything you do with them would produce the same result.  The only thing that would be different is explicitly accessing the `noise` attribute.  

In other words, there is nothing in GalSim that uses the `noise` attribute of an object.  It's only there for convenience of the user if they want to do something with it.  So if the user cares, they can explicitly check the equality of two objects' `noise` attributes.  And the only way they could possibly end up different is for the user to have explicitly added or deleted it from one of them somewhere along the way.